src/start/RealmScreen.js

```diff
@@ -33,8 +33,22 @@ class RealmScreen extends React.Component {
   }
 
   tryRealm = async () => {
-    const { realm } = this.state;
-
+    const protocol = /:\/\//;
+    let { realm } = this.state;
+    if (protocol.test(realm)) {
+      const tokens = realm.split(protocol);
+      switch (tokens[0]) {
+        case ('http'):
+          break;
+        case ('https'):
+          break;
+        default:
+          this.setState({ progress: false, error: 'Unsupported protocol' });
```

Comment:
You don't need to set progress to false here.
Not sure if explicitly handling the protocol as separate error is worth it.